pkg/controllers/osc/osc_controller.go

```diff
@@ -152,12 +168,24 @@ func (r *Reconciler) reconcileOperatingSystemConfigs(ctx context.Context, md *cl
 		return fmt.Errorf("failed to get OperatingSystemProfile: %v", err)
 	}
 
+	secret := &corev1.Secret{}
+	if err := r.Get(ctx, types.NamespacedName{Name: resources.CloudConfigSecretName, Namespace: metav1.NamespaceSystem}, secret); err != nil {
```

Comment:
You probably need to do this only for cloud providers which actually need a cloud config. E.g: DO, Hetzner, etc..)

Reply:
As far as I see from the Machine-controller code, the cloud-config is always rendered: https://github.com/kubermatic/machine-controller/blob/master/pkg/userdata/ubuntu/provider.go#L257-L260. By default it is the cloud-config fetched from the machine, and if the `overwriteCloudConfig` != nil, then the `cloud-config` is overwritten by it.

Reply:
The cloud-configs are fetched from the machine deployment, that's correct, problem is, in your implementation this is would lead to a lot of issues. For example, which service is gonna create this secret for you? If the values are already in the machine why reading out of secret? why not simply do like what machine controller is doing which is reading and parsing these configs from the machine deployment, since you have already an access to it during reconciliation. Imagine this scenario, a DO cluster is provisioned, in that case osc controller will simply fail to generate any provisioning configs because it will keep failing with object not found error because DO doesn't have cloud config